src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotUseStackallocInLoops.cs

```diff
@@ -39,8 +39,18 @@ public sealed override void Initialize(AnalysisContext context)
                 // We found a stackalloc.  Walk up from it to see if it's in a loop at any level.
                 for (SyntaxNode node = ctx.Node; node != null; node = node.Parent)
                 {
+                    // Don't warn of the stackalloc is in anonymous function.
+                    if (ctx.SemanticModel.GetOperation(node, ctx.CancellationToken)?.Kind == OperationKind.AnonymousFunction)
```

Comment:
Sorry I have read too fast earlier! I think it's better not to go for the `SemanticModel.GetOperation` and so to have 2 syntax kind exceptions on the switch.